test/auth/auth_mock_test.go

```diff
@@ -285,3 +286,28 @@ func TestAuthMock_CheckDefaultScopes(t *testing.T) {
 	AuthErr := client.Schema().AllDeleter().Do(context.TODO())
 	assert.Nil(t, AuthErr)
 }
+
+// Test that the correct header is set when using an API Key to authenticate.
+func TestWithSimpleAuthNoOidcViaApiKey(t *testing.T) {
+	token := "super-secret-key"
+	mux := http.NewServeMux()
+	mux.HandleFunc("/v1/schema", func(w http.ResponseWriter, r *http.Request) {
+		authHeader := r.Header.Get("Authorization")
+		require.Equal(t, authHeader, "Bearer "+token)
+		w.Write([]byte(`{}`))
+	})
+	mux.HandleFunc("/v1/.well-known/ready", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`{}`))
+	})
+	s := httptest.NewServer(mux)
+	defer s.Close()
+
+	headers := map[string]string{}
+	cfg := weaviate.Config{Host: strings.TrimPrefix(s.URL, "http://"), Scheme: "http", Headers: headers}
+	var err error
+	cfg, err = weaviate.AddAuthClient(cfg, auth.ApiKeys{ApiKey: token}, 60*time.Second)
+	assert.Nil(t, err)
+	client := weaviate.New(cfg)
+	AuthErr := client.Schema().AllDeleter().Do(context.TODO())
+	assert.Nil(t, AuthErr)
+}
```

Comment:
This should also be for `weaviate.AddAuthClient` and `NewConfig`.

You can simplify this by doing

```
func TestWithSimpleAuthNoOidcViaApiKey(t *testing.T) {
	token := "super-secret-key"
	mux := http.NewServeMux()
	mux.HandleFunc("/v1/schema", func(w http.ResponseWriter, r *http.Request) {
		authHeader := r.Header.Get("Authorization")
		require.Equal(t, authHeader, "Bearer "+token)
		w.Write([]byte(`{}`))
	})
	mux.HandleFunc("/v1/.well-known/ready", func(w http.ResponseWriter, r *http.Request) {
		w.Write([]byte(`{}`))
	})
	s := httptest.NewServer(mux)
	defer s.Close()
       
       t.Run(tc.name, func(t *testing.T) {
	    headers := map[string]string{}
	    cfg := weaviate.Config{Host: strings.TrimPrefix(s.URL, "http://"), Scheme: "http", Headers: headers}
	    var err error
	    cfg, err = weaviate.AddAuthClient(cfg, auth.ApiKeys{ApiKey: token}, 60*time.Second)
	    assert.Nil(t, err)
	    client := weaviate.New(cfg)
	    AuthErr := client.Schema().AllDeleter().Do(context.TODO())
	    assert.Nil(t, AuthErr)
       })

       t.Run(tc.name, func(t *testing.T) {
        -- code for NewConfig----
       })
}
```